crates/bevy_ecs/src/query/state.rs

```diff
@@ -132,7 +135,7 @@ impl<D: QueryData, F: QueryFilter> QueryState<D, F> {
     /// `NewD` must have a subset of the access that `D` does and match the exact same archetypes/tables
     /// `NewF` must have a subset of the access that `F` does and match the exact same archetypes/tables
     pub(crate) unsafe fn as_transmuted_state<
-        NewD: QueryData<State = D::State>,
+        NewD: ReadOnlyQueryData<State = D::State>,
```

Comment:
Why do we restrict this to ReadOnlyQueryData?

Reply:
I suppose if this is a problem in the future, we can deal with it then! So I'm fine with this.